### ISSUE TITLE
Document (figure out) more specific purpose of "available"

### ISSUE THREAD

**Issue by davegaeddert** [dependencies-io/schema#5]
Should it be *all* versions? Probably not, as this isn't even achievable if you consider commit revisions and other possibilities.  

A thought...

For collectors the "available" versions should be:
- if semver is installed, include semver versions above the currently installed version
- if not using semver, include all semver versions available
- if using a branch or alias, include refs that are above the current ref (or latest ref only if that's all that can be easily found)
- if using a specific ref...?
- if it is transitive, only include versions matching constraint (available lock file updates)

In summary, we should only report the versions we think the user would be interested in.

In all cases, those that are in-range vs out-of-range should be marked (#2) which would make it more clear what their relationship is to the installed version/constraint.

The actors use of available stays the same (the versions it acted on). Or the actor should start using a slimmed down / different schema to better match its usage.

**Comment by davegaeddert** [comment]
Much more clear in v2.
